perfkitbenchmarker/virtual_machine.py

```diff
@@ -138,6 +150,9 @@ def __init__(self, vm_spec):
     self._total_memory_kb = None
     self._num_cpus = None
 
+    self._has_remote_command_script_lock = threading.Lock()
```

Comment:
`_has_remote_command_script_lock` protects the boolean value `_has_remote_command_script`. Naming here was to be consistent with `_instance_counter_lock` / `_instance_counter`.  I'm worried `_remote_command_script_lock` will imply that a single remote command should be run at a time, which is not the case. 

Renamed to `_remote_command_script_upload_lock`.